### PR TITLE
Updates for wolfEngine

### PR DESCRIPTION
**Pull request by ejohnstown**
Added a directory for open source projects that needed minor modifications to use wolfEngine under OpenSSL for compatibility.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ Each project port included in this repository is contained in its own subdirecto
 | urllib3 | urllib3 HTTP client for Python | [Link](https://github.com/urllib3/urllib3) | | [README](./urllib3/README.rst) |
 | websocket-client | WebSocket client for python | [Link](https://github.com/websocket-client/websocket-client) | | [README](./websocket-client/README.rst) |
 | websocketpp | WebSocket++ | [Link](https://www.zaphoyd.com/projects/websocketpp/) | [Link](https://www.wolfssl.com/building-websocket-wolfssl-support/) | [README](websocketpp/readme.md) |
+| wolfengine | OpenSSL Engine using wolfCrypt | [Link](https://www.wolfssl.com/) | | [README](./wolfengine/README.md) |
```
Comment (haydenroche5) [inline README.md:54]: Would prefer a link to the wolfEngine GitHub.

File: wolfengine/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+The goal of the wolfEngine project is to replace the crypto used by OpenSSL
+with wolfCrypt or wolfCrypt FIPS, with the least amount of changes possible.
+
+Each of the project directories has both a patch file and a reference to a
+submodule for the project at the commit for the patch. After cloning the OSP
+repository, change directories into the project's subdirectory. Run the
+command `git submodule update && git submodule update`. Or if the OSP
+repository hasn't been cloned yet, clone it with the command:
+
+    git clone --recurse-submodules https://github.com/wolfSSL/wolfEngine.git
```
Comment (haydenroche5) [inline wolfengine/README.md:12]: This isn't the OSP URL.

File: wolfengine/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# wolfSSL Open Source Project Ports with wolfEngine
+
+The goal of the wolfEngine project is to replace the crypto used by OpenSSL
+with wolfCrypt or wolfCrypt FIPS, with the least amount of changes possible.
+
+Each of the project directories has both a patch file and a reference to a
+submodule for the project at the commit for the patch. After cloning the OSP
+repository, change directories into the project's subdirectory. Run the
+command `git submodule update && git submodule update`. Or if the OSP
```
Comment (haydenroche5) [inline wolfengine/README.md:9]: I think you mean `git submodule init && git submodule update`. For some reason, when I pull down this PR as a patch and apply it, the submodule commands aren't doing anything:

```
hayden@hayden-dell: /home/hayden/wolfssl_osp/wolfengine/libssh$ ls -l
total 8
---------- 1 hayden hayden  59 Jan  4 07:41 libssh
-rw-r--r-- 1 hayden hayden 624 Jan  4 07:41 libssh-patch.txt
hayden@hayden-dell: /home/hayden/wolfssl_osp/wolfengine/libssh$ git submodule init && git submodule update
hayden@hayden-dell: /home/hayden/wolfssl_osp/wolfengine/libssh$ ls -l
total 8
---------- 1 hayden hayden  59 Jan  4 07:41 libssh
-rw-r--r-- 1 hayden hayden 624 Jan  4 07:41 libssh-patch.txt
```

File: wolfengine/libssh/libssh-patch.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+commit 2dbd0f035a82a7423cf6688b385f47cd52b8ad2f
```
Comment (haydenroche5) [inline wolfengine/libssh/libssh-patch.txt:1]: Please change directory name wolfengine to wolfEngine.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: wolfengine/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+repository hasn't been cloned yet, clone it with the command:
+
+    git clone --recurse-submodules https://github.com/wolfSSL/wolfEngine.git
+
+Change directories into the project's directory and run
+`patch -p1 ../PROJ-patch.txt` to apply the patch file. At this point, build
```
Comment (lealem47) [inline wolfengine/README.md:15]: Missing `<` in patch command .  `patch -p1 < ../PROJ-patch.txt`

**Review by lealem47 — CHANGES_REQUESTED** [review]

File: wolfengine/README.md (hunk 3 above)
Comment (lealem47) [reply wolfengine/README.md:9]: @haydenroche5 `git submodule init && git submodule update` should be run from the inner-most project directory. The wording in the instructions could be changed to make that more clear.

Comment (haydenroche5) [reply wolfengine/README.md:9]: I don't think that's the problem:

Comment (lealem47) [reply wolfengine/README.md:9]: Oh, I see the same thing when applying the PR as a patch. Not sure why it's not working but fetching then checking out to the PR worked fine

File: README.md (hunk 1 above)
Comment (ejohnstown) [reply README.md]: Done.

File: wolfengine/README.md (hunk 2 above)
Comment (ejohnstown) [reply wolfengine/README.md]: Fixed.

File: wolfengine/README.md (hunk 3 above)
Comment (ejohnstown) [reply wolfengine/README.md]: I updated the directions there. It should have been init then update, but you do it anywhere in the repo directory it updates them all.

It looks like you have to actually check out the project or branch. If you do it with a patch it doesn't work. Just inserting a `.gitmodules` file to a repo doesn't seem to do the submodule setup.

File: wolfengine/libssh/libssh-patch.txt (hunk 4 above)
Comment (ejohnstown) [reply wolfengine/libssh/libssh-patch.txt]: Done.

File: wolfengine/README.md (hunk 5 above)
Comment (ejohnstown) [reply wolfengine/README.md]: Got it, thanks.

**Review by lealem47 — APPROVED** [review]

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**